core/src/main/scala/tech/beshu/ror/boot/engines/BaseReloadableEngine.scala

```diff
@@ -189,9 +190,14 @@ private[engines] abstract class BaseReloadableEngine(val name: String,
 
   private def stop(engineWithConfig: EngineWithConfig)
                   (implicit requestId: RequestId): Unit = {
+    onStop()
     engineWithConfig.engine.shutdown()
     logger.info(s"[${requestId.show}] ROR $name engine (id=${engineWithConfig.config.hashString()}) stopped!")
   }
+
+  protected def onSuccessLoad(c: EngineWithConfig): Unit
```

Comment:
I'm not sure if we have to define these two callback methods. Please, take a look at how we extract `engine` from  `currentEngine` State. It looks like we can extract raw config in the same way. Moreover, we have also info that the engine was not started yet or was stopped.